lang/go/bindings/examples/distributor/pkg/distributor/tx.go

```diff
@@ -0,0 +1,92 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright(c) 2022-2023 Intel Corporation
+ */
+
+package distributor
+
+import (
+	"context"
+	"sync"
+	"unsafe"
+
+	"github.com/CloudNativeDataPlane/cndp/lang/go/bindings/cne"
+)
+
+type TX struct {
+	ctx    context.Context
+	handle *cne.System
+	wg     sync.WaitGroup
+	ring   *cne.LocklessRing
+}
+
+func NewTX(ctx context.Context, handle *cne.System, lb LoadBalancer) (*TX, error) {
+	ring, err := cne.NewLRing("tx-ring", initialBufferSize, []string{cne.RingFlagSingleConsumer})
+	if err != nil {
+		return nil, err
+	}
+	tx := &TX{
+		ctx:    ctx,
+		handle: handle,
+		ring:   ring,
+	}
+	lb.SetCallBack(tx.Aggregate)
+
+	return tx, nil
+}
+
+func (tx *TX) send(thdName string, lportNames []string) {
+	lports := tx.handle.LPortsByName(lportNames)
+	if len(lports) == 0 {
+		return
+	}
+
+	err := tx.handle.RegisterThread(thdName)
+	if err != nil {
+		return
+	}
+	defer tx.handle.UnregisterThread(thdName)
+
+	items := make([]uintptr, numPackets)
+
+	var lportIds []int
+	for _, lport := range lports {
+		lportIds = append(lportIds, lport.LPortID())
+	}
+
+	tx.wg.Add(1)
+	for {
+		for _, pid := range lportIds {
+			select {
+			case <-tx.ctx.Done():
+				tx.wg.Done()
+				return
+			default:
+				n := tx.ring.DequeueN(items, numPackets)
+
+				for _, item := range items[:n] {
+					packet := (*cne.Packet)(unsafe.Pointer(item))
+
+					cne.TxBurst(pid, []*cne.Packet{packet}, true)
```

Comment:
The TxBurst() routine returns N or zero if the packets can not be sent. If a zero is returned the link is down and should we continue looping forever here? I think we need to handle the zero return case, maybe a message if we are in this loop for too long, because the link is down.

Reply:
Updated, and will dump an error message if the loop takes more than 1 sec.